app/components/spotlight/analytics/dashboard_component.rb

```diff
@@ -9,7 +9,47 @@ class DashboardComponent < ViewComponent::Base
       def initialize(current_exhibit:)
         super
         @current_exhibit = current_exhibit
-        @dates = { 'start_date' => '365daysAgo', 'end_date' => 'today' }
+        @default_start_date = 1.year.ago.to_date.to_s
+        @default_end_date = Time.zone.today.to_date.to_s
+      end
+
+      def before_render
+        flash[:error] = nil
+        @dates = { 'start_date' => date(:start_date, @default_start_date),
+                   'end_date' => date(:end_date, @default_end_date) }
+      end
+
+      def date(param_var, default_date)
+        param_value = params[param_var]
+        return default_date unless param_value
+
+        begin
+          Date.parse(param_value)
+          param_value
+        rescue StandardError
```

Comment:
Any reason not to rescue the more specific `Date::Error` instead?

Reply:
The other error case I'm seeing is the date is parseable but it's outside the allowable range from GA:

```
Google::Cloud::InvalidArgumentError
3:start_date = 9999-10-10 must be greater than 2015-08-13 and less than 3000-01-01..
```